### ISSUE TITLE
CI: Remove building from Unit Test-Jobs

### ISSUE THREAD

**Issue by marcoabreu** [apache/incubator-mxnet#8726]
At the moment the compile-stages are kept pretty generic while language-specific builds are being executed during the Unit Test-Stage. In order to improve future scalability these two steps should be two separate jobs. This can be done in two ways:

1. Add an second build-stage which executes language-specific actions. Disadvantage: This increases the total duration as all jobs have to finish until the next stage can be started.
2. Nested jobs: If a Unit Test requires previous actions, split it into two jobs which are getting executed in sequence. This will allow in future to execute CPU-heavy tasks on a CPU-instance and GPU-tests on a GPU-instance.

Execution in separate jobs allows reusing the created artifacts if multiple jobs exist for the same language.
For examples consider looking in the Jenkinsfile at R: CPU/GPU, Scala: CPU and Perl: CPU/GPU (perl-package.test.sh).

**Comment by marcoabreu** [comment]
Also consider splitting up tests like tests/python/unittest and tests/python/train in order to allow parallel execution.

**Comment by cjolivier01** [comment]
@KellenSunderland

**Comment by piyushghai** [comment]
@marcoabreu I believe our existing Jenkins setup does this separation of unit test jobs for different language bindings. 
Can this issue be closed then ?

**Comment by marcoabreu** [comment]
This is still valid. Languages like scala, clojure, R and probably a few more compile their native counterparts and dependencies during unit test stage instead of compile stage.

**Comment by piyushghai** [comment]
Aah. I see. Thanks!
